src/seat.c

```diff
@@ -952,14 +952,25 @@ find_session_config (Seat *seat, const gchar *sessions_dir, const gchar *session
     g_return_val_if_fail (sessions_dir != NULL, NULL);
     g_return_val_if_fail (session_name != NULL, NULL);
 
+    g_auto(GStrv) session_name_type;
```

Comment:
Instead of using `session_name_type[1]` later on and `g_strv_length` can you make a `session_type` variables that is NULL if not specified? It will make the code a bit clearer.